### PR TITLE
Outline zhang shasha

### PR DESCRIPTION
**Pull request by snim2**
Outline of a module for the Zhang-Shasha tree matching algorithm. Some small refactorings for readability / remove code duplication.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -285,8 +285,8 @@ fn process_doc_flags(args: &Args) {
 /// Will cause diffract to exit if user-requested lexer / parser does not exist.
 fn get_parsers(args: &Args) -> (PathBuf, PathBuf, PathBuf, PathBuf) {
     // TODO: create a HashMap of file extensions -> lex/yacc files.
-    match env::current_exe() {
-        Ok(_) => (),
+     match env::current_exe() {
```
Comment (ltratt) [inline src/main.rs]: Not sure that space was meant to be introduced?

**Comment by ltratt** [comment]
I wonder if we can find a nicer function name than `same_type_and_label`? Maybe not though.

**Comment by snim2** [comment]
Let me have a think about the fn names...

**Comment by snim2** [comment]
OK, ready to squash?

**Comment by snim2** [comment]
These new names are long, but at least it's clear what the functions do!

**Comment by ltratt** [comment]
OK, if you can squash and rebase, I can merge.

**Comment by snim2** [comment]
OK, done...

**Pull request merged**